atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -144,31 +164,86 @@ public static SerializableTransactionManager create(
+    }
+
+    /**
+     * Create a {@link SerializableTransactionManager} with provided install configuration, a set of
+     * {@link Schema}s, {@link LockServerOptions}, and an environment in which to register HTTP server endpoints.
+     *
+     * @deprecated Use mirror method with runtime configuration parameter.
+     */
+    @Deprecated
+    public static SerializableTransactionManager create(
+            AtlasDbConfig config,
+            Set<Schema> schemas,
+            Environment env,
+            LockServerOptions lockServerOptions,
+            boolean allowHiddenTableAccess) {
+        return create(config, () -> AtlasDbRuntimeConfig.create(config),
+                schemas, env, lockServerOptions, allowHiddenTableAccess, UserAgents.DEFAULT_USER_AGENT);
+    }
+
+    /**
+     * Create a {@link SerializableTransactionManager} with provided configurations, a set of
      * {@link Schema}s, {@link LockServerOptions}, and an environment in which to register HTTP server endpoints.
      */
     public static SerializableTransactionManager create(
             AtlasDbConfig config,
+            java.util.function.Supplier<AtlasDbRuntimeConfig> runtimeConfig,
             Set<Schema> schemas,
             Environment env,
             LockServerOptions lockServerOptions,
             boolean allowHiddenTableAccess) {
-        return create(config, schemas, env, lockServerOptions, allowHiddenTableAccess, UserAgents.DEFAULT_USER_AGENT);
+        validateRuntimeConfig(runtimeConfig.get());
+        return create(config, runtimeConfig, schemas, env, lockServerOptions, allowHiddenTableAccess,
+                UserAgents.DEFAULT_USER_AGENT);
+    }
+
+    /**
+     * @deprecated Use mirror method with runtime configuration parameter.
+     */
+    @Deprecated
+    public static SerializableTransactionManager create(
+            AtlasDbConfig config,
+            Set<Schema> schemas,
+            Environment env,
+            LockServerOptions lockServerOptions,
+            boolean allowHiddenTableAccess,
+            Class<?> callingClass) {
+        return create(config, () -> AtlasDbRuntimeConfig.create(config),
+                schemas, env, lockServerOptions, allowHiddenTableAccess,
+                UserAgents.fromClass(callingClass));
     }
 
     public static SerializableTransactionManager create(
             AtlasDbConfig config,
+            java.util.function.Supplier<AtlasDbRuntimeConfig> runtimeConfig,
             Set<Schema> schemas,
             Environment env,
             LockServerOptions lockServerOptions,
             boolean allowHiddenTableAccess,
             Class<?> callingClass) {
-        return create(config, schemas, env, lockServerOptions, allowHiddenTableAccess,
+        validateRuntimeConfig(runtimeConfig.get());
+        return create(config, runtimeConfig, schemas, env, lockServerOptions, allowHiddenTableAccess,
```

Comment:
The purpose if the log is to make people who manage a stack that supports runtime configs actually add them.
While the name of the method denotes that the runtime config will be validated, i.e. at each run will be validated and we'll abort if there's something invalid, this was not its purpose. I cannot think of a better name though.